lib/imap.c

```diff
@@ -660,9 +660,27 @@ static CURLcode imap_perform_list(struct Curl_easy *data)
     free(mailbox);
   }
 
-  if(!result)
+  if(!result) {
     state(data, IMAP_LIST);
 
+    /* Switch state to be FETCH if custom command is a fetch command */
+    /* so output is handled correctly. */
+
+    if(imap->custom) {
+      if(strncasecompare(imap->custom, "FETCH", 5) == 1) {
+        state(data, IMAP_FETCH);
+      }
+
+      /* e.g. "UID FETCH 107 BODY.PEEK[]" */
+
+      if(imap->custom_params)
+      if(strncasecompare(imap->custom, "UID", 3) == 1)
+      if(strncasecompare(imap->custom_params, " FETCH", 6) == 1) {
+        state(data, IMAP_FETCH);
+      }
```

Comment:
1.This code doesn't use the proper indenting, two spaces for new levels.
2.I don't think it should change state if it immediately again changes to something else. Change only once.

Reply:
I added spaces for intending.
I think setting IMAP_FETCH twice should not be a problem.
it is much easier code to do it this way than to create one huge if to handle both cases together.

But feel free to adjust if needed.

Reply:
The problem is primarily that it makes it harder to understand what's going on when you look at state transitions in a verbose log but also that it isn't actually changing the state if you immediately change it again so it's sort of a lie.

If we keep state changes to *actual* state changes, we keep the ability to at some point add state init logic there, the way we've done in some of our other state machines.

Reply:
We could of course add "return result; " after changing state to fetch.
And then do the IMAP_LIST if the compares didn't set it to fetch.

Reply:
I committed a new suggestion for the implementation with an isFetch flag, which only changes state once now.
Comments?